src/scripts/match-mondo-sources-all-lexical.py

```diff
@@ -85,11 +102,105 @@ def run(input:str, config: str, rules:str, output: str):
     
     # msdf.prefix_map = sssom_yaml['curie_map']
     # msdf.metadata = sssom_yaml['global_metadata']
+    msdf.df = filter_prefixes(df=msdf.df, filter_prefixes=prefix_of_interest, features=[SUBJECT_ID, OBJECT_ID])
+    # TODO: replace line below by msdf.remove_mappings(mapping_msdf) once imported from SSSOM.
+    msdf.df = (
+            pd.merge(
+                msdf.df,
+                mapping_msdf.df,
+                on=KEY_FEATURES,
+                how="outer",
+                suffixes=("", "_2"),
+                indicator=True,
+            )
+            .query("_merge == 'left_only'")
+            .drop("_merge", axis=1)
+            .reset_index(drop=True)
+        )
+    msdf.df = msdf.df[msdf.df.columns.drop(list(msdf.df.filter(regex=r"_2")))]
+    msdf.clean_prefix_map()
 
     with open(str(SRC / Path(output)), "w", encoding="utf8") as f:
         write_table(msdf, f)
 
+    objects = msdf.df[OBJECT_ID].drop_duplicates()
+    prefixes = objects.str.split(":").str.get(0).drop_duplicates()
+    prefix_args = tuple([x+":%" for _, x in prefixes.iteritems() if x != "MONDO"])
+    kwargs = {"subject_id": ("MONDO:%",), "object_id": prefix_args}
+    with open(str(SRC / Path(output.replace("lexical", "lexical-2"))), "w") as f:
+        filter_file(
+            input=str(SRC / Path(output)),
+            output=f,
+            **kwargs
+            )
+
+# TODO: Replace the method below by sssom.utils.filter_prefixes.
```

Comment:
Maybe if there are these two case (and the remove_mappings one above), lets keep the PR open and make an sssom release to fix this. Just to be clean, we are not in a super rush now.

Reply:
I agree!